src/netex-convertor/handler.ts

```diff
@@ -61,68 +59,80 @@ const uploadToS3 = async (netex: string, fileName: string): Promise<void> => {
 
 export const generateFileName = (eventFileName: string): string => eventFileName.replace('.json', '.xml');
 
+export const buildNocList = (ticket: PointToPointTicket | PeriodTicket | SchemeOperatorTicket): string[] => {
+    const nocs: string[] = [];
+
+    // if (isPointToPointTicket(ticket)) {
+    //     nocs.push(ticket.nocCode);
+    // } else if (isPeriodTicket(ticket)) {
+    //     if (ticket.type === 'multiOperator') {
+    //         if (isMultiOperatorGeoZoneTicket(ticket) || isSchemeOperatorTicket(ticket)) {
+    //             nocs.concat(ticket.additionalNocs);
+    //             if (isMultiOperatorGeoZoneTicket(ticket)) {
+    //                 nocs.push(ticket.nocCode);
+    //             }
+    //         } else if (isMultiOperatorMultipleServicesTicket(ticket)) {
+    //             const additionalNocs = ticket.additionalOperators.map(additionalOperator => additionalOperator.nocCode);
+    //             nocs.concat(additionalNocs);
+    //         } else {
+    //             nocs.push(ticket.nocCode);
+    //         }
+    //     } else if (
+    //         !isMultiOperatorGeoZoneTicket(ticket) &&
+    //         !isMultiOperatorMultipleServicesTicket(ticket) &&
+    //         !isSchemeOperatorTicket(ticket)
+    //     ) {
+    //         nocs.push(ticket.nocCode);
+    //     }
+    // }
+
+    if (isPointToPointTicket(ticket)) {
+        nocs.push(ticket.nocCode);
+    } else if (isPeriodTicket(ticket)) {
+        const userPeriodTicket: PeriodTicket = ticket;
+        if (ticket.type === 'multiOperator') {
+            if (isMultiOperatorGeoZoneTicket(userPeriodTicket) || isSchemeOperatorTicket(userPeriodTicket)) {
+                const additionalNocs: string[] = [...userPeriodTicket.additionalNocs];
+                if (isMultiOperatorGeoZoneTicket(userPeriodTicket)) {
+                    additionalNocs.push(userPeriodTicket.nocCode);
+                }
+                additionalNocs.forEach(additionalNoc => nocs.push(additionalNoc));
+            } else if (isMultiOperatorMultipleServicesTicket(userPeriodTicket)) {
+                const additionalOperatorNocs: string[] = userPeriodTicket.additionalOperators.map(
+                    additionalOperator => additionalOperator.nocCode,
+                );
+                additionalOperatorNocs.push(userPeriodTicket.nocCode);
+                additionalOperatorNocs.forEach(additionalOperatorNoc => nocs.push(additionalOperatorNoc));
+            }
+        } else if (
+            !isMultiOperatorGeoZoneTicket(userPeriodTicket) &&
+            !isMultiOperatorMultipleServicesTicket(userPeriodTicket) &&
+            !isSchemeOperatorTicket(userPeriodTicket)
```

Comment:
Because the type guards only get called from line 94 onwards inside the if multi operator block, they arent always called. Which means that we need to use the typeguards again here just for typescript to be happy that its not those types.